diagnostics/model_test/verify_model.py

```diff
@@ -61,47 +61,85 @@ def calculate_output(param_dict, select_device, input_example):
         The output vector obtained from running the input_example through the graph.
     """
     graph_filename = param_dict["graph_location"]
+    checkpoint_filename = param_dict["checkpoint_location"]
     output_tensor_name = param_dict["output_tensor_name"]
 
-    if not tf.gfile.Exists(graph_filename):
-        raise Exception("Input graph file '" + graph_filename +
-                        "' does not exist!")
+    config = tf.ConfigProto(
+        # log_device_placement=True,
+        inter_op_parallelism_threads=1)
 
-    graph_def = tf.GraphDef()
-    if graph_filename.endswith("pbtxt"):
-        with open(graph_filename, "r") as f:
-            text_format.Merge(f.read(), graph_def)
-    else:
-        with open(graph_filename, "rb") as f:
-            graph_def.ParseFromString(f.read())
+    sess = tf.Session(config=config)
+    import pdb
+    #pdb.set_trace()
+    if tf.train.checkpoint_exists(
+            checkpoint_filename) and not tf.gfile.Exists(graph_filename):
+        is_ckpt = True
+    if tf.gfile.Exists(
+            graph_filename) and not tf.gfile.Exists(checkpoint_filename):
+        is_ckpt = False
+    if not tf.train.checkpoint_exists(
+            checkpoint_filename) and not tf.gfile.Exists(graph_filename):
+        raise Exception("Input graph file '" + graph_filename + "' or" +
+                        "Input checkpoint file '" + checkpoint_filename +
+                        "' does not exist!")
+    #pdb.set_trace()
 
     set_os_env(select_device)
 
-    with tf.Graph().as_default() as graph:
-        tf.import_graph_def(graph_def)
+    # if checkpoint, then load checkpoint
+    if (is_ckpt):
+        saver = tf.train.import_meta_graph(checkpoint_filename + '.meta')
+        saver.restore(sess, checkpoint_filename)
+        print("Model restored.")
+        graph_from_ckpt = tf.get_default_graph()
         if len(output_tensor_name) == 0:
             # if no outputs are specified, then compare for all tensors
             output_tensor_name = sum(
-                [[j.name for j in i.outputs] for i in graph.get_operations()],
-                [])
+                [[j.name
+                  for j in i.outputs]
+                 for i in graph_from_ckpt.get_operations()], [])
+
+    if tf.gfile.Exists(graph_filename):
+        graph_def = tf.GraphDef()
+        if graph_filename.endswith("pbtxt"):
+            with open(graph_filename, "r") as f:
+                text_format.Merge(f.read(), graph_def)
+        else:
+            with open(graph_filename, "rb") as f:
+                graph_def.ParseFromString(f.read())
+
+        with tf.Graph().as_default() as graph:
+            tf.import_graph_def(graph_def)
+            if len(output_tensor_name) == 0:
+                # if no outputs are specified, then compare for all tensors
+                output_tensor_name = sum([
+                    [j.name for j in i.outputs] for i in graph.get_operations()
+                ], [])
```

Comment:
Is there any difference between `graph` and  `graph_from_ckpt`?
If not please refactor the `output_tensor_name` computation outside the if-else